From 3fe25e69dca3949b40ab6ff4882de2c4ee1c3312 Mon Sep 17 00:00:00 2001
From: Tristan <tde@hey.com>
Date: Thu, 20 Apr 2023 09:02:38 +0100
Subject: [PATCH] Test resetting signlaal handler

---
 src/haproxy.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/haproxy.c b/src/haproxy.c
index fd43c963a..8aa11f607 100644
--- a/src/haproxy.c
+++ b/src/haproxy.c
@@ -3200,6 +3200,8 @@ int main(int argc, char **argv)
 	int pidfd = -1;
 	int intovf = (unsigned char)argc + 1; /* let the compiler know it's strictly positive */
 
+	signal(SIGSEGV, SIG_DFL);
+
 	/* Catch broken toolchains */
 	if (sizeof(long) != sizeof(void *) || (intovf + 0x7FFFFFFF >= intovf)) {
 		const char *msg;
-- 
2.40.0

